code/game/machinery/chem_heater.dm

```diff
@@ -0,0 +1,113 @@
+/obj/machinery/chem_heater
+	name = "chemical heater"
+	density = 1
+	anchored = 1
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "mixer0b" //GET IN THE ICON
+	use_power = 1
+	idle_power_usage = 300
+	var/obj/item/weapon/reagent_containers/beaker = null
+	var/desired_temp = 300
+	var/heater_coefficient = 0.10
+	var/on = 0
+
+/obj/machinery/chem_heater/process()
+	..()
+	if(stat & NOPOWER)
+		return
+	if(on)
+		if(beaker)
+			if(beaker.reagents.chem_temp > desired_temp)
+				beaker.reagents.chem_temp += min(-1, (desired_temp - beaker.reagents.chem_temp) * heater_coefficient)
+			if(beaker.reagents.chem_temp < desired_temp)
+				beaker.reagents.chem_temp += max(1, (desired_temp - beaker.reagents.chem_temp) * heater_coefficient)
+			beaker.reagents.chem_temp = round(beaker.reagents.chem_temp)
+
+			beaker.reagents.handle_reactions()
+
+/obj/machinery/chem_heater/power_change()
+	if(powered())
+		stat &= ~NOPOWER
+	else
+		spawn(rand(0, 15))
+			stat |= NOPOWER
+
+/obj/machinery/chem_heater/attackby(obj/item/I, mob/user, params)
+	if(isrobot(user))
+		return
+
+	if(istype(I, /obj/item/weapon/reagent_containers/glass))
+		if(beaker)
+			user << "<span class='warning'>A beaker is already loaded into the machine!</span>"
+			return
+
+		if(user.drop_item())
+			beaker = I
+			I.loc = src
+			user << "<span class='notice'>You add the beaker to the machine.</span>"
+			icon_state = "mixer1b"
+
+/obj/machinery/chem_heater/attack_hand(mob/user)
+	if (!user)
+		return
+	interact(user)
+
+//Artifact from /tg/
+/*/obj/machinery/chem_heater/ui_act(action, params)
```

Comment:
Hah that's tgui, tgui's new UI system. You could just make this `Topic()` instead...